engine/consensus/approvals/approval_collector.go

```diff
@@ -83,15 +106,21 @@ func (c *ApprovalCollector) SealResult() error {
 	}
 
 	// we don't care if the seal is already in the mempool
-	_, err = c.seals.Add(&flow.IncorporatedResultSeal{
+	added, err := c.seals.Add(&flow.IncorporatedResultSeal{
 		IncorporatedResult: c.incorporatedResult,
 		Seal:               seal,
 		Header:             c.executedBlock,
 	})
 	if err != nil {
 		return fmt.Errorf("failed to store IncorporatedResultSeal in mempool: %w", err)
 	}
-
+	if added {
+		c.log.Info().
+			Str("executed_block_id", seal.BlockID.String()).
+			Str("result_id", seal.ResultID.String()).
+			Str("incorporating_block", c.IncorporatedBlockID().String()).
+			Msg("added candidate seal to IncorporatedResultSeals mempool")
```

Comment:
Good suggestion: added block height to log. 

> ask mempool whether that seal exists

💯 The `SealingRecord` does this: https://github.com/onflow/flow-go/blob/791cf86012ad429fac4ebe8f3ec13c7873d30b5d/engine/consensus/approvals/tracker/record.go#L88

Nevertheless, I think that _additionally_ logging when we first add the seal to the mempool might be useful in case there is a sealing halt, where the next needed is missing. Without this log, we would not de able to distinguish whether 
* the seal was added and later ejected 
* the seal was never added in the first place.

Reply:
makes sense 👍